src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -1355,6 +1357,67 @@ TEST_CASE("LedgerTxn load", "[ledgertxn]")
                         NonSociRelatedException);
                 }
 
+                SECTION("invalid asset")
+                {
+                    std::string accountIDStr, issuerStr, assetCodeStr;
+                    auto loadTest = [&](Asset const& asset) {
+                        auto key = trustlineKey(acc2.getPublicKey(), asset);
+
+                        // verify that this doesn't throw before V15
+                        getTrustLineStrings(key.trustLine().accountID,
+                                            key.trustLine().asset, accountIDStr,
+                                            issuerStr, assetCodeStr, 14);
+
+                        REQUIRE_THROWS_AS(ltx1.load(key),
+                                          NonSociRelatedException);
+                    };
+
+                    SECTION("control char in asset name")
+                    {
+                        auto asset = txtest::makeAsset(acc, "\n");
+                        loadTest(asset);
+                    }
+                    SECTION("non-trailing zero in asset name")
+                    {
+                        std::string assetCode;
+                        assetCode.push_back(0);
+                        assetCode.push_back('a');
+                        auto asset = txtest::makeAsset(acc, assetCode);
+                        loadTest(asset);
+                    }
+                    SECTION("zero asset name")
+                    {
+                        std::string assetCode;
+                        assetCode.push_back(0);
+                        auto asset = txtest::makeAsset(acc, assetCode);
+                        loadTest(asset);
+                    }
+                    SECTION("invalid ascii chars")
+                    {
+                        // start right after z(122), and go through some of the
+                        // extended ascii codes
+                        for (int i = 123; i < 140; ++i)
+                        {
+                            std::string assetCode;
+                            assetCode.push_back(i);
+                            SECTION(fmt::format("asset name={}", assetCode))
```

Comment:
Nitpick: this might not be the best idea, since it won't be easy to specify any of these sections directly. Could either remove the sections entirely, or do something that references the ASCII number instead of the printed char. I'm not that picky about this regardless, since I don't anticipate ever wanting to run just one of these sections.

Reply:
yeah I agree: this `SECTION` just makes this test case very slow as it forces to redo the setup.

I would suggest removing this section, and most of the other ones as well as they don't have side effect (so they can safely be done one after the other) see `INFO` and `UNSCOPED_INFO` to add context when there are failures.

Reply:
Fixed